### PR TITLE
feat: shadowSupportMode=native

### PR DESCRIPTION
**Pull request by gonzalocordero**
## Details

Introduces renderMode = native for semantic rendering of native only components

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js [hunk 1]
```diff
@@ -40,3 +41,26 @@ describe('ENABLE_MIXED_SHADOW_MODE', () => {
         setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
     });
 });
+
+describe('ENABLE_NATIVE_SHADOW_MODE', () => {
+    beforeEach(() => {
+        setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', true);
+    });
```
Comment (ekashida) [inline packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:48]: Native shadow should be used without this feature flag so we don't need to enable here.

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js [hunk 2]
```diff
@@ -40,3 +41,26 @@ describe('ENABLE_MIXED_SHADOW_MODE', () => {
...
+
+    it('should be configured as "any" (sanity)', () => {
```
Comment (ekashida) [inline packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:50]:
```suggestion
    it('should be configured as "native" (sanity)', () => {
```

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js [hunk 3]
```diff
@@ -40,3 +41,26 @@ describe('ENABLE_MIXED_SHADOW_MODE', () => {
...
+        expect(NativeOnly.shadowSupportMode === 'native').toBeTrue();
+    });
+
+    it('should enable mixed shadow mode', () => {
```
Comment (ekashida) [inline packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:54]:
```suggestion
    it('should render native shadow root', () => {
```

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js [hunk 4]
```diff
@@ -40,3 +41,26 @@ describe('ENABLE_MIXED_SHADOW_MODE', () => {
...
+        const elm = createElement('x-native-only', { is: NativeOnly });
+        if (process.env.NATIVE_SHADOW_ROOT_DEFINED) {
+            expect(isNativeShadowRootInstance(elm.shadowRoot)).toBeTrue();
+        } else {
+            expect(isSyntheticShadowRootInstance(elm.shadowRoot)).toBeTrue();
```
Comment (ekashida) [inline packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:59]: I guess the expectation here is that an exception is thrown? I'm not even sure we test with browsers that don't implement shadow roots natively.

File: packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js [hunk 5]
```diff
@@ -40,3 +41,26 @@ describe('ENABLE_MIXED_SHADOW_MODE', () => {
...
+        }
+    });
+
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
+    });
```
Comment (ekashida) [inline packages/@lwc/integration-karma/test/mixed-shadow-mode/LightningElement.shadowSupportMode/index.spec.js:65]: No need for this as explained above.

**Comment by nolanlawson** [comment]
Superseded by https://github.com/salesforce/lwc/pull/3771

**Pull request closed without merging**